File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;

class Solution {
    public int[] solution(int n, String[] words) {
        Set<String> set = new HashSet<>();
        char pre = words[0].charAt(words[0].length() - 1);
        set.add(words[0]);
        
        for (int i = 1; i < words.length; i++) {
            if (set.contains(words[i]) || words[i].charAt(0) != pre) {
                int num = (i % n) + 1;
                int round = (i / n) + 1;
                
                return new int[]{num, round};
            }
            set.add(words[i]);
            pre = words[i].charAt(words[i].length() - 1);
        }
        return new int[]{0, 0};
    }
}

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.js
function solution(n, words) {
    let set = new Set();
    set.add(words[0]);
    
    for (let i = 1; i < words.length; i++) {
        if (set.has(words[i]) || 
            words[i - 1].charAt(words[i - 1].length - 1) !== words[i].charAt(0)
           ) {
            return [Math.floor(i % n + 1), Math.floor(i / n + 1)];
        }
        set.add(words[i]);
    }
    return [0, 0];
}

// 8